src/iris/api.py

```diff
@@ -3959,11 +3959,14 @@ def on_get(self, req, resp):
 class ApplicationStats(object):
     allow_read_no_auth = True
 
-    def on_get(self, req, resp, app_name):
-        app = cache.applications.get(app_name)
-        if not app:
-            raise HTTPNotFound()
+    def __init__(self, config):
+        cfg = config.get('app-stats', {})
+        # Calculate stats in real time (True), or query offline stats
+        # generated by the app stats daemon (False)
+        self.real_time = cfg.get('real_time', True)
 
+    @staticmethod
+    def calculate_app_stats(app, connection, cursor, fields_filter=None):
```

Comment:
Can we move this calculate_app_stats function to somewhere outside the Falcon route class? It's kind of odd for a daemon script to call a method that is part of a route